numcodecs/checksum32.py

```diff
@@ -1,39 +1,75 @@
 import struct
 import zlib
+from typing import Literal
 
 import numpy as np
+from crc32c import crc32c as crc32c_
 
 from .abc import Codec
 from .compat import ensure_contiguous_ndarray, ndarray_copy
 from .jenkins import jenkins_lookup3
 
+CHECKSUM_LOCATION = Literal['start', 'end']
+
 
 class Checksum32(Codec):
     # override in sub-class
     checksum = None
+    location: CHECKSUM_LOCATION = 'start'
```

Comment:
Would be good to add a quick docstring in `__init__` to describe the new parameter and attribute.

Reply:
I added it to the class docstring. Unfortunately not very DRY. Not sure if docstrings can be dynamic?